app/controllers/posts_controller.rb

```diff
@@ -12,7 +12,8 @@ def show
   end
 
   def new
-    @post = Post.new
+    @user = current_user

```

Comment:
- Add CanCanCan authorization to post controller by adding `load_and_authorize_resource` before the actions.